reference/src/taimpl/src/internal/symmetric.c

```diff
@@ -1542,16 +1542,16 @@ bool symmetric_context_check_tag(
     if (EVP_CIPHER_CTX_ctrl(context->evp_cipher, EVP_CTRL_AEAD_SET_TAG, (int) tag_length, (void*) tag) != 1) {
 #endif
         ERROR("EVP_CIPHER_CTX_ctrl failed");
-        return false;
+        return SA_STATUS_INTERNAL_ERROR;
     }
 
     int length = 0;
     if (EVP_DecryptFinal_ex(context->evp_cipher, NULL, &length) != 1) {
         ERROR("EVP_DecryptFinal_ex failed");
-        return false;
+        return SA_STATUS_INVALID_PARAMETER;
```

Comment:
Is this an invalid parameter or internal error condition?

Reply:
It is an invalid parameter. In this case the tag is wrong. Documented accordingly.